### PR TITLE
Recursively resolve thrift includes in dependencies

### PR DESCRIPTION
**Pull request by aengelberg**
@greglook

### CODE REVIEW

File: lein-finagle-clojure/src/leiningen/finagle_clojure.clj [hunk 1]
```diff
@@ -65,14 +65,28 @@
...
+    (loop [q (into clojure.lang.PersistentQueue/EMPTY external-deps)
+           result []]
+      (if (empty? q)
+        result
+        (let [filename (peek q)
+              {:keys [content]} (thrift-dependencies filename)]
+          (recur (into (pop q)
+                       (when content
+                         ;; If content is nil, we have a problem, but
+                         ;; the scrooge task will abort accordingly
```
Comment (greglook) [inline lein-finagle-clojure/src/leiningen/finagle_clojure.clj:87]: Curious why you use two semicolons for these embedded comments?

File: lein-finagle-clojure/src/leiningen/finagle_clojure.clj [hunk 2]
```diff
@@ -65,14 +65,28 @@
...
         relative-sources (->> source-paths
                               (map #(subs % (inc (count (str root)))))
-                              (set))]
-    (set/difference include-paths relative-sources)))
+                              (set))
+        external-deps (set/difference include-paths relative-sources)]
+    ;; For each of those external dependencies, recursively scrape
+    ;; THEM to find any transitive dependencies. Breadth-first search.
+    (loop [q (into clojure.lang.PersistentQueue/EMPTY external-deps)
+           result []]
```
Comment (greglook) [inline lein-finagle-clojure/src/leiningen/finagle_clojure.clj]: Seems like `result` should be a set and you should not re-enqueue scraped includes which are already part of the set.

File: lein-finagle-clojure/src/leiningen/finagle_clojure.clj (hunk 1 above)
Comment (aengelberg) [inline lein-finagle-clojure/src/leiningen/finagle_clojure.clj:87]: https://github.com/bbatsov/clojure-style-guide#comments

File: lein-finagle-clojure/src/leiningen/finagle_clojure.clj (hunk 2 above)
Comment (aengelberg) [inline lein-finagle-clojure/src/leiningen/finagle_clojure.clj]: Ah, good point

File: lein-finagle-clojure/src/leiningen/finagle_clojure.clj (hunk 1 above)
Comment (greglook) [inline lein-finagle-clojure/src/leiningen/finagle_clojure.clj:87]: Fair enough!

**Comment by greglook** [comment]
Changes LGTM! :shipit:

**Pull request merged**